src/core/types/message.rs

```diff
@@ -4,17 +4,177 @@ use crate::core::types::{
     SuccessfulPayment, User, Venue, Video, VideoNote, Voice,
 };
 
-#[derive(Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+#[derive(Debug, Deserialize, Eq, Hash, PartialEq)]
 pub struct Message {
     #[serde(rename = "message_id")]
-    pub id: i32,
+    pub id: i64,
     pub date: i32,
-    pub chat: Chat,
+    pub chat: Box<Chat>,
     #[serde(flatten)]
-    pub type_: MessageType,
+    pub message_kind: MessageKind,
 }
 
-#[derive(Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
-#[serde(rename_all = "snake_case")]
+#[derive(Debug, Deserialize, Eq, Hash, PartialEq)]
 #[serde(untagged)]
-pub enum MessageType {}
+pub enum MessageKind {
+    IncomingMessage {
+        #[serde(flatten)]
+        forward_kind: ForwardKind,
+        edit_date: Option<i32>,
+        #[serde(flatten)]
+        media_kind: MediaKind,
+        reply_markup: Option<InlineKeyboardMarkup>,
+    },
+    NewChatMembers {
+        new_chat_members: Vec<User>,
+    },
+    LeftChatMember {
+        left_chat_member: User,
+    },
+    NewChatTitle {
+        new_chat_title: String,
+    },
+    NewChatPhoto {
+        new_chat_photo: Vec<PhotoSize>,
+    },
+    DeleteChatPhoto {
+        delete_chat_photo: bool,
+    },
+    GroupChatCreated {
+        group_chat_created: bool,
+    },
+    SupergroupChatCreated {
+        supergroup_chat_created: bool,
+    },
+    ChannelChatCreated {
+        channel_chat_created: bool,
+    },
+    Migrate {
+        migrate_to_chat_id: i64,
+        migrate_from_chat_id: i64,
+    },
+    PinnedMessage {
+        pinned_message: Box<Message>,
+    },
+    Invoice {
+        invoice: Invoice,
+    },
+    SuccessfulPayment {
+        successful_payment: SuccessfulPayment,
+    },
+    ConnectedWebsite {
+        connected_website: String,
+    },
+    PassportData {
+        passport_data: PassportData,
+    },
+}
+
+#[derive(Debug, Deserialize, Eq, Hash, PartialEq)]
+#[serde(untagged)]
+pub enum ForwardKind {
+    ChannelForward {
+        forward_from_channel: Box<Chat>,
+        forward_from_message_id: i64,
+        forward_signature: String,
+    },
+    NonChannelForward {
+        #[serde(flatten)]
+        forward: UserOrSenderName,
+    },
+    Origin {
+        reply_to_message: Option<Box<Message>>,
+    },
+}
+
+#[derive(Debug, Deserialize, Eq, Hash, PartialEq)]
+#[serde(untagged)]
+pub enum UserOrSenderName {
```

Comment:
Maybe rename it to `ForwardedFrom`?...
```suggestion
pub enum ForwardedFrom {
```